spec/models/keyword_recommendation_spec.rb

```diff
@@ -0,0 +1,44 @@
+
+# cannot create recommendation by itself
+# draft can create a recommendation
+# template can create a recommendation
+
+# in proposal mode, proposal does not create a recommendation ?????
+
+describe KeywordRecommendation do
```

Comment:
Did you consider a test for cascade delete? I'm leaning towards "it's testing rails", but am open to others' thoughts.

Reply:
I did not consider testing it, but it might be a good idea. I think it is also 'testing rails' a bit, but couldn't hurt. It should probably go in the other side of the associations though.

Reply:
added tests